hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -52,6 +53,8 @@
 
     Collection<R> values();
 
+    Iterator<R> recordIterator(boolean failFast);
```

Comment:
Documentation needs to be added what kind of iterator this is.  So does it support concurrent behavior or not. And what does fail fast mean? This needs to be documented as well.

Why not call this method 'iterator' since it is obvious that from a record store you would get an iterator of records.